solace-message-router/nestedtemplates/loadbalancer-shared-resources.json

```diff
@@ -0,0 +1,280 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "dnsLabelForLbIp": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Label for the Public IP used to access the Load Balancer."
+      }
+    },
+    "publicIPAddressNameLB": {
+      "type": "string",
+      "defaultValue": "myLBPublicIPD",
+      "metadata": {
+        "description": "Load Balancer Public IP Address Name."
+      }
+    },
+    "publicIPAddressType": {
+      "type": "string",
+      "defaultValue": "Dynamic",
+      "metadata": {
+        "description": "Load Balancer Public IP Address Type."
+      }
+    },
+    "lbName": {
+      "type": "string",
+      "defaultValue": "myLB",
+      "metadata": {
+        "description": "Load Balancer Name."
+      }
+    },
+    "lbPoolName": {
+      "type": "string",
+      "defaultValue": "solace-ha-group",
+      "metadata": {
+        "description": "Load Balancer Backend Address Pool Name."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "variables": {
+    "lbID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'))]",
+    "frontEndIPConfigName": "LoadBalancerFrontEnd",
+    "frontEndIPConfigID": "[concat(variables('lbID'),'/frontendIPConfigurations/',variables('frontEndIPConfigName'))]",
```

Comment:
use resourceId for the ids

Reply:
Need a bit of help how to use resourceId for the ids.

Tried:
```
    "lbID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'))]",
    "frontEndIPConfigName": "LoadBalancerFrontEnd",
    "frontEndIPConfigID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'),'frontendIPConfigurations',variables('frontEndIPConfigName'))]",
    "lbPoolID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'),'backendAddressPools',parameters('lbPoolName'))]",
    "lbProbeName": "solace-ha-ad-health-check",
    "lbProbeID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'),'probes',variables('lbProbeName'))]"
```

Got:
```
"error": {
    "code": "InvalidTemplate",
    "message": "Deployment template validation failed: 'The template variable 'frontEndIPConfigID' is not valid: Unable to evaluate template language function 'resourceId': the type 'Microsoft.Network/loadBalancers' requires '1' resource name argument(s). Please see https://aka.ms/arm-template-expressions/#resourceid for usage details.. Please see https://aka.ms/arm-template-expressions for usage details.'."
  }
```

Reply:
it's a multi-part resourceID, so

resourceId('Microsoft.Network/loadbalances/frontendIPConfigurations', parameters('lbName'), variables('frontEndIPConfigName'))